src/csv/lexer.cr

```diff
@@ -89,15 +89,15 @@ abstract class CSV::Lexer
       when '\0'
         raise "unclosed quote"
         break
-      when '"'
+      when @quote_char
         case next_char
-        when ','
+        when @separator
           check_last_empty_column
           break
         when '\r', '\n', '\0'
           break
-        when '"'
-          @buffer << '"'
+        when @quote_char
```

Comment:
you mean that i should store it in a local variable before usage? I don't think it's an issue here since there is no setter as you pointed out. What you described might happen if you have concurrent code working on the same instance (and with possibilities to change the value).